test/tools/runner/hooks/configuration.ts

```diff
@@ -211,8 +213,37 @@ const beforeAllPluginImports = () => {
   require('mocha-sinon');
 };
 
+async function beforeEachLogging(this: Context) {
+  if (this.currentTest == null) return;
+  this.configuration.beforeEachLogging(this);
+}
+
+async function afterEachLogging(this: Context) {
+  if (this.currentTest == null) return;
+  this.configuration.afterEachLogging(this);
+}
+
+function checkFlakyTestList(this: Context) {
+  const allTests: string[] = [];
+
+  const stack = [this.test.parent];
+  while (stack.length) {
+    const suite = stack.pop();
+    allTests.push(...suite.tests.map(test => test.fullTitle()));
+    stack.push(...suite.suites);
+  }
+  allTests.reverse(); // Doesn't matter but when debugging easier to see this in the expected order.
+
+  const flakyTestDoesNotExist = flakyTests.find(testName => !allTests.includes(testName));
```

Comment:
This isn't erroring for me locally.  I added a new test to the list of flaky tests that doesn't exist ('asdf'), and then ran the tests, but no error was reported

Reply:
Looks like this now fails non-regular test run environments

Reply:
Dang, I was worried about that I only thought mongocryptd would be the issue but we were fixing that. I supposed I can use the environment config to exempt the assertion 🤔 but that could also be brittle. Should we just print a warning, and we'll catch it eventually (maybe not _on_ the PR that removes a test)

Reply:
I can't think of a way to do this without some sort of brittle solution.  

Printing a warning sounds fine

Reply:
changed it to a print with some extra noise to help catch our eye